src/lib/errors.ts

```diff
@@ -35,3 +39,45 @@ export const getRouteErrorMessageOverride = (
     return routeErrorMessage;
   }
 };
+
+const getUntranslatedErrorMessageOrDefaultErrorParams = (errorMessage?: string): ErrorParams => {
+  if (errorMessage && errorMessage !== '') return { errorMessage };
+  return DEFAULT_SOMETHING_WENT_WRONG_ERROR_PARAMS;
+};
+
+export const getValidParsingErrorParams = (error?: Nullable<ParsingError>): ErrorParams => {
+  const { stringKey: errorStringKey, message: errorMessage } = error ?? {};
+  const defaultErrorParams = getUntranslatedErrorMessageOrDefaultErrorParams(errorMessage);
+  if (!errorStringKey) return defaultErrorParams;
+
+  const validErrorStringKey = STRING_KEY_VALUES[errorStringKey];
+  if (!validErrorStringKey) {
+    const err = new Error(`Missing error translation for ${errorStringKey}`);
+    log('errors/MissingParsingErrorTranslation', err, { parsingError: error });
+    return defaultErrorParams;
+  }
+
+  return { errorStringKey: validErrorStringKey };
+};
+
+/**
+ * Abacus parses the stringified error and returns a ParseError.
+ * BroadcastError is parsed with `code`, `codespace`, and `message`, and defaults to show the original error message if untranslated.
+ * Failed Query results (i.e., those starting with "Query failed") are matched via the error message, defaulting to "Unknown query result error".
+ * Unmatched errors will display the actual error message.
+ */
+export const stringifyTransactionError = (error: any): string => {
```

Comment:
[q] does broadcasterror cover all abacus cases? i'm curious as to whether or not this will also fix the onboarding/offboarding modal errors

Reply:
[this](https://github.com/dydxprotocol/v4-abacus/blob/11d7c881307d9c252111d35c93e7bc90bbcde44d/src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/NetworkHelper.kt#L528) is where transaction errors are parsed!

that specific error in the slack channel is a broadcast error, so i think it's just because it wasn't passed through that function